### PR TITLE
Use Java 7 APIs and language features

### PR DESCRIPTION
**Pull request by gaul**
Found via error-prone and Modernizer.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -317,6 +317,23 @@
           </dependency>
         </dependencies>
       </plugin>
+      <plugin>
+        <groupId>org.gaul</groupId>
+        <artifactId>modernizer-maven-plugin</artifactId>
```
Comment (eolivelli) [inline pom.xml]: I am not sure we want this plugin and the dependency.
I see it is an useful tool.

Maybe we should start a thread on dev@maven.apache.org

@rfscholte

Comment (Tibor17) [reply pom.xml]: @gaul 
hehe, a good trick ;-)
I will copy this config for myself, but you should really remove it here.

**Comment by Tibor17** [comment]
@gaul 
You don't want to add the commit from #17 ?
I think it's all the Java 7 stuff.

File: pom.xml (hunk 1 above)
Comment (gaul) [reply pom.xml]: Done.

**Comment by gaul** [comment]
I think of #17 as separate since it is language vs. APIs but I can combine them if you prefer.

**Comment by Tibor17** [comment]
@gaul We made the same changes in Surefire due to the changes in API are introduced in J7. So yes, the J7 has something new in language and API too but it is acceptable by Java compiler.

**Comment by gaul** [comment]
Done.

**Comment by gaul** [comment]
Addressed all outstanding suggestions.

**Comment by Tibor17** [comment]
LGTM
This work is worth a Jira ticket.
@gaul 
You will find the Jira URL in the POM. Pls login there and publish a new ticket with the same name of this PR. Then it would be a complete work.
Thx, and it's a good job in real!

**Comment by Tibor17** [comment]
Jira ticket created https://issues.apache.org/jira/browse/MPLUGIN-354
Let's check this fix on Jenkins tomorrow.

**Comment by gaul** [comment]
I created [MPLUGIN-353](https://issues.apache.org/jira/browse/MPLUGIN-353) earlier.

**Comment by Tibor17** [comment]
@gaul 
ok, i will delete mine. Can you rewrite the commit message to:
`[MPLUGIN-353] Use Java 7 APIs and language features`
Thx

**Comment by Tibor17** [comment]
@gaul 
@eolivelli 
The build is currently running on Jenkins https://builds.apache.org/job/maven-box/job/maven-plugin-tools/job/MPLUGIN-353/

**Comment by Tibor17** [comment]
@gaul 
I am going to push this PR to master. If you want us to cut a new release version `3.6.1` force us on the mailing list for developer. Meanwhile we will wait for new fixes.

**Comment by Tibor17** [comment]
closing as merged

**Pull request closed without merging**